expertsystem/solving/__init__.py

```diff
@@ -439,42 +499,105 @@
+    initialized_graphs = []
+
+    logging.debug("merging solutions with graph...")
+    intermediate_edges = graph.get_intermediate_state_edges()
+    for solution in solutions:
+        temp_graph = deepcopy(graph)
+        for node_id in temp_graph.nodes:
+            if node_id in solution.node_quantum_numbers:
+                temp_graph.node_props[node_id] = solution.node_quantum_numbers[
+                    node_id
+                ]
+
+        current_new_graphs = [temp_graph]
+        for int_edge_id in intermediate_edges:
+            particle_edges = __get_particle_candidates_for_state(
+                solution.edge_quantum_numbers[int_edge_id],
+                allowed_particles,
+            )
+            if len(particle_edges) == 0:
+                logging.debug("Did not find any particle candidates for")
+                logging.debug("edge id: %d", int_edge_id)
+                logging.debug("edge properties:")
+                logging.debug(solution.edge_quantum_numbers[int_edge_id])
+            new_graphs_temp = []
+            for current_new_graph in current_new_graphs:
+                for particle_edge in particle_edges:
+                    temp_graph = deepcopy(current_new_graph)
+                    temp_graph.edge_props[int_edge_id] = particle_edge
+                    new_graphs_temp.append(temp_graph)
+            current_new_graphs = new_graphs_temp
+
+        initialized_graphs.extend(current_new_graphs)
+
+    return initialized_graphs
+
+
+def __get_particle_candidates_for_state(
+    state: Dict[Type[EdgeQuantumNumber], Union[int, float]],
+    allowed_particles: ParticleCollection,
+) -> List[ParticleWithSpin]:
+    particle_edges: List[ParticleWithSpin] = []
+
+    for allowed_particle in allowed_particles.values():
+        if __check_qns_equal(state, allowed_particle):
+            # temp_particle = deepcopy(allowed_particle)
+            particle_edges.append(
+                (allowed_particle, state[EdgeQuantumNumbers.spin_projection])
+            )
+
+    return particle_edges
+
+
+def __check_qns_equal(
+    state: Dict[Type[EdgeQuantumNumber], Union[int, float]], particle: Particle
+) -> bool:
+    # This function assumes the attribute names of Particle and the quantum
+    # numbers defined by new type match
+    changes_dict: Dict[str, Union[int, float, Parity, Spin]] = {
+        getattr(edge_qn, "__name__"): value
+        for edge_qn, value in state.items()
+        if "magnitude" not in getattr(edge_qn, "__name__")
+        and "projection" not in getattr(edge_qn, "__name__")
+    }
+
+    if EdgeQuantumNumbers.isospin_magnitude in state:
+        changes_dict["isospin"] = Spin(
+            state[EdgeQuantumNumbers.isospin_magnitude],
+            state[EdgeQuantumNumbers.isospin_projection],
+        )
+    if EdgeQuantumNumbers.spin_magnitude in state:
+        changes_dict["spin"] = state[EdgeQuantumNumbers.spin_magnitude]
+    return replace(particle, **changes_dict) == particle
```

Comment:
Interesting, probably also good to use this in the implementation of `create_(anti_)particle`.

Reply:
I thought its used there already, at least I remember I pointed this out before

Reply:
Can be handled in #294